### ISSUE TITLE
Selection of the best hyper parameters for the UNET Model for semantic segmentation.

### ISSUE THREAD

**Issue by saiphanish7** [yassouali/pytorch_segmentation#52]
I would like to know what is the best optimizer to use on a ( UNET Model, PASCAL VOC Dataset for segmentation, Focal loss ) or is it preferable to use an lr scheduler method. Can you please give the optimizer to work with and it's parameters?

I have tried using Adam ( lr = 0.0002 , beta1 =0.5, beta2 = 0.999 ) but the mean_iou is not increasing rather, it is increasing and decreasing but it's almost same after every epoch on my validation set of Pascal VOC for image segmentation.

**Comment by yassouali** [comment]
For semantic segmentation, SGM with momentum is generally used, and the learning rate schedule of choice is POLY learning rage, where we start from LR_base and linearity decrease down to

**Comment by saiphanish7** [comment]
I first want to test my standard UNET Model on Pascal VOC using MseLoss for semantic segmentation. Is it fine to use SGD with LR_base 0.01 , Weight decay 1e-4, momentum 0.9 and poly learning rate scheduler ?
If that works , then I will test it using focal loss.

**Comment by saiphanish7** [comment]
Can you please provide the results of mean_iou on pascal voc val set using UNET model and optimizer specifications for it. please?

**Comment by yassouali** [comment]
Sorry, I am kinda busy tight now, I can't train the model.  For unet, you need higher learning rate given that you're training from scratch, try 0.1.

**Comment by yassouali** [comment]
Please refer to this issue, we'll try to post the training results for other models. https://github.com/yassouali/pytorch_segmentation/issues/29

**Comment by saiphanish7** [comment]
If possible, please train unet model with focal loss on Pascal Voc and share the results.🙂

**Comment by saiphanish7** [comment]
If possible, please train unet model with focal loss on Pascal Voc and share the results.🙂

**Comment by saiphanish7** [comment]
As you told, I have trained the unet model on SGD optimizer with 0.1 LR and poly learning rate policy but the results are poor.

**Comment by yassouali** [comment]
I'll get back to you in about 2/3 weeks.

**Comment by saiphanish7** [comment]
Can you tell me whether to use val.txt or train.txt for Pascal Voc validation because some of the implementations used train.txt for validation.

**Comment by saiphanish7** [comment]
> I'll get back to you in about 2/3 weeks.

Can you please give any update regarding the issue?

**Comment by sravanjosh07** [comment]
Hey @saiphanish7 , have you achieved what you wanted?
